### ISSUE TITLE
Numpy 1.11.0 under Python 2.6 requires nose not only for tests

### ISSUE THREAD

**Issue by hannosch** [numpy/numpy#7498]
A production install of numpy 1.11.0 suddenly requires nose, even though this isn't mentioned in install_requires. This only happens under Python 2.6, likely due to some unittest backports for the SkipTest decorator.

This was most likely accidentally introduced in a61ddd3812cc95f9c9e6eeac7f8bcfb92130f978.

Steps to reproduce:
```
$ virtualenv --python=python2.6 foo
$ cd foo/
$ bin/pip install numpy==1.11.0
Collecting numpy==1.11.0
Installing collected packages: numpy
Successfully installed numpy-1.11.0

$ bin/python -c "import numpy.core"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/__init__.py", line 180, in <module>
    from . import add_newdocs
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/lib/__init__.py", line 8, in <module>
    from .type_check import *
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/lib/type_check.py", line 11, in <module>
    import numpy.core.numeric as _nx
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/core/__init__.py", line 58, in <module>
    from numpy.testing.nosetester import _numpy_tester
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/testing/__init__.py", line 12, in <module>
    from . import decorators as dec
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/testing/decorators.py", line 21, in <module>
    from .utils import SkipTest
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/testing/utils.py", line 50, in <module>
    SkipTest = import_nose().SkipTest
  File "/opt/tmp/foo/lib/python2.6/site-packages/numpy/testing/nosetester.py", line 73, in import_nose
    raise ImportError(msg)
ImportError: Need nose >= 1.0.0 for tests - see http://somethingaboutorange.com/mrl/projects/nose
```

**Comment by charris** [comment]
@njsmith What is the easiest fix here? The problem commit could be reverted for 1.11 as that is the last release to support Python 2.6.

**Comment by njsmith** [comment]
Have we confirmed that that commit is the problem? Looking at it I can't immediately see how it affects anything related to nose imports...

**Comment by charris** [comment]
@njsmith Good point. Actually, it looks like 70c5052a9, #6688.  @ev-br That was yours. Hmm, we could make a local copy if needed.

**Comment by charris** [comment]
The problem is
```
try:
    from unittest.case import SkipTest
except ImportError:
    # on py2.6 unittest.case is not available. Ask nose for a replacement.
    SkipTest = import_nose().SkipTest
```

**Comment by charris** [comment]
And the problem arises in `numpy/core/__init__.py`
```
from numpy.testing.nosetester import _numpy_tester
test = _numpy_tester().test
bench = _numpy_tester().bench
```
@njsmith Those are yours from a61ddd38 as reported.

**Comment by njsmith** [comment]
@charris: right, but I just swapped one way of creating a `NoseTester` for a different way of creating it, and the problem is that `import numpy.testing` errors out...

It looks like the quick fix would be (in maintenance-1.11 only) modify `numpy/testing/utils.py` to do
```
# nose.SkipTest is unittest.case.SkipTest
# import it into the namespace, so that it's available as np.testing.SkipTest
try:
    from unittest.case import SkipTest
except ImportError:
    # on py2.6 unittest.case is not available. Ask nose for a replacement.
    # this code gets executed unconditionally, so we can't depend on nose being available
    # if it's not, oh well, something else will issue an error when people try to actually run tests
    try:
        import nose
    except ImportError:
        pass
    else:
        SkipTest = nose.SkipTest
```

and on master we can just make it
```
from unittest.case import SkipTest
```

**Comment by charris** [comment]
Closed by #7506.

**Comment by aragilar** [comment]
This is still broken. If nose isn't installed, SkipTest cannot be imported. See https://travis-ci.org/aragilar/numpy-test-py2.6/builds/144072784 for an example. I guess this could be fixed by adding nose to install_requires in numpy's setup.py, but I suspect that's not a popular option. Maybe needing nose should be added to the installation instructions?

**Comment by njsmith** [comment]
Yeah, it looks like the change here: https://github.com/numpy/numpy/pull/7506/files
made it so `numpy.testing.utils` skips importing `SkipTest` if `nose` isn't available, but then `numpy.testing.decorators` does `from .utils import SkipTest`, which fails because when `nose` is missing then the fallback in `numpy.testing.utils` doesn't bind anything to the `SkipTest` name at all. I guess addin ga `SkipTest = None` on the `except` block would work.

@aragilar: if you want to submit a PR against the maintenance/1.11.x branch, that'd be welcome, though I'm not 100% sure whether there will be any more 1.11.x releases anyway.

**Comment by aragilar** [comment]
I don't use python 2.6, I was just trying to track down why h5py's tests were failing. Given I'd need to have a working python 2.6 install to debug this (I used travis as that was the fastest way I could think of running something against 2.6), I'm not planning on submitting a pull request.

**Comment by charris** [comment]
Should fix this for 1.11.2.
